src/Stratis.ModuleValidation.Net/TypeHasNestedTypesValidator.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil;
+
+namespace Stratis.ModuleValidation.Net
+{
+    public class TypeHasNestedTypesValidator : ITypeDefinitionValidator 
```

Comment:
docs - We know what a 'nested type' is but for others might be nice to know that it includes internal structs, enums + classes.

Reply:
Those are own own rules about what's allowed, but this validator checks for the presence of any nested types. I assume most people will already have an understanding of what nested types are.